platform/darwin/src/local_glyph_rasterizer.mm

```diff
@@ -45,47 +46,72 @@ CTFontDescriptor configuration option (although we'd have to override font size
 using CGContextHandle = CFHandle<CGContextRef, CGContextRef, CGContextRelease>;
 using CFStringRefHandle = CFHandle<CFStringRef, CFTypeRef, CFRelease>;
 using CFAttributedStringRefHandle = CFHandle<CFAttributedStringRef, CFTypeRef, CFRelease>;
+using CFMutableArrayRefHandle = CFHandle<CFMutableArrayRef, CFTypeRef, CFRelease>;
 using CFDictionaryRefHandle = CFHandle<CFDictionaryRef, CFTypeRef, CFRelease>;
+using CTFontRefHandle = CFHandle<CTFontRef, CFTypeRef, CFRelease>;
 using CTFontDescriptorRefHandle = CFHandle<CTFontDescriptorRef, CFTypeRef, CFRelease>;
 using CTLineRefHandle = CFHandle<CTLineRef, CFTypeRef, CFRelease>;
 
 class LocalGlyphRasterizer::Impl {
 public:
     Impl(const optional<std::string> fontFamily_)
-        : fontFamily(fontFamily_)
-        , fontHandle(NULL)
-    {}
+    {
+        fallbackFontNames = [[NSUserDefaults standardUserDefaults] stringArrayForKey:@"MGLIdeographicFontFamilyName"];
+        if (fontFamily_) {
+            fallbackFontNames = [fallbackFontNames ?: @[] arrayByAddingObjectsFromArray:[@(fontFamily_->c_str()) componentsSeparatedByString:@"\n"]];
+        }
+    }
     
     ~Impl() {
-        if (fontHandle) {
-            CFRelease(fontHandle);
-        }
     }
-
     
-    CTFontRef getFont() {
-        if (!fontFamily) {
-            return NULL;
+    bool isEnabled() { return fallbackFontNames; }
+    
+    CTFontDescriptorRef createFontDescriptor(const FontStack& fontStack) {
+        NSMutableArray *fontNames = [NSMutableArray arrayWithCapacity:fontStack.size() + fallbackFontNames.count];
+        for (auto& fontName : fontStack) {
+            if (fontName != util::LAST_RESORT_ALPHABETIC_FONT && fontName != util::LAST_RESORT_PAN_UNICODE_FONT) {
```

Comment:
Are `LAST_RESORT_ALPHABETIC_FONT` and `LAST_RESORT_PAN_UNICODE_FONT` private to gl-native?

Reply:
This PR adds those two constants to constants.hpp for use anywhere in mbgl, including this file.